common/src/main/java/org/jboss/pnc/common/monitor/PullingMonitor.java

```diff
@@ -49,15 +52,32 @@
     /** */
     private static final TimeUnit DEFAULT_TIME_UNIT = TimeUnit.SECONDS;
 
+    private static final String PULLING_MONITOR_THREADPOOL_KEY = "pulling-monitor-threadpool";
+    private static final int DEFAULT_EXECUTOR_THREADPOOL_SIZE = 4;
+
     private ScheduledExecutorService executorService;
     private ScheduledExecutorService timeOutVerifierService;
 
     private ConcurrentSet<RunningTask> runningTasks;
 
     public PullingMonitor() {
+
+        int threadSize = DEFAULT_EXECUTOR_THREADPOOL_SIZE;
+        String threadSizeEnv = System.getProperty(PULLING_MONITOR_THREADPOOL_KEY);
+
```

Comment:
Please add an option to read the value also from environment variable "PNC_PULLING_MONITOR_THREADPOOL".
Where system property overrides env when both are set.